06_puppy/sahidakhtar1/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+type Puppy struct {
+	ID                   uint
+	Breed, Colour, Value string
+}
+
+type Storer interface {
+	CreatePuppy(Puppy)
+	ReadPuppy(uint) Puppy
+	UpdatePuppy(uint, Puppy)
+	DeletePuppy(uint) bool
+}
+
+//Map Store Starts
+type MapStore struct {
+	m map[uint]Puppy
+}
+
+func newMapStore() *MapStore {
+	ms := MapStore{}
+	ms.m = make(map[uint]Puppy)
+	return &ms
+}
+func (ms *MapStore) CreatePuppy(p Puppy) {
+	_, ok := ms.m[p.ID]
```

Comment:
Create should take care of `ID`.
Consider adding a `maxID uint` field to `MapStore` and using here like this:
```go
ms.maxID++
p.ID = maxID
ms.m[p.ID] = p
```
This is not thread/go routine save, but it will suffice as a "toy" implementation.

Reply:
Done